cmd/booster-bitswap/client.go

```diff
@@ -0,0 +1,220 @@
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"net/http"
+	_ "net/http/pprof"
+	"sort"
+	"sync/atomic"
+	"time"
+
+	"github.com/filecoin-project/boost/cmd/booster-bitswap/bitswap"
+	lcli "github.com/filecoin-project/lotus/cli"
+	"github.com/ipfs/go-bitswap/client"
+	bsnetwork "github.com/ipfs/go-bitswap/network"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	nilrouting "github.com/ipfs/go-ipfs-routing/none"
+	ipldlegacy "github.com/ipfs/go-ipld-legacy"
+	"github.com/ipld/go-car/v2/blockstore"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/muxer/mplex"
+	"github.com/libp2p/go-libp2p/p2p/muxer/yamux"
+	quic "github.com/libp2p/go-libp2p/p2p/transport/quic"
+	"github.com/libp2p/go-libp2p/p2p/transport/tcp"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/sync/errgroup"
+)
+
+var fetchCmd = &cli.Command{
+	Name:        "fetch",
+	Usage:       "fetch <multiaddr> <root cid> <output car path>",
+	Description: "Fetch all blocks in the DAG under the given root cid from the bitswap node at multiaddr",
+	Before:      before,
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name:  "pprof",
+			Usage: "run pprof web server on localhost:6071",
+		},
+		&cli.IntFlag{
+			Name:  "concurrency",
+			Usage: "concurrent request limit - 0 means unlimited",
+			Value: 10,
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		if cctx.Bool("pprof") {
+			go func() {
+				err := http.ListenAndServe("localhost:6071", nil)
+				if err != nil {
+					log.Error(err)
+				}
+			}()
+		}
+
+		if cctx.Args().Len() != 3 {
+			return fmt.Errorf("usage: fetch <multiaddr> <root cid> <output car path>")
+		}
+
+		addrInfoStr := cctx.Args().Get(0)
+		serverAddrInfo, err := peer.AddrInfoFromString(addrInfoStr)
+		if err != nil {
+			return fmt.Errorf("parsing server multiaddr %s: %w", addrInfoStr, err)
+		}
+
+		rootCidStr := cctx.Args().Get(1)
+		rootCid, err := cid.Parse(rootCidStr)
+		if err != nil {
+			return fmt.Errorf("parsing cid %s: %w", rootCidStr, err)
+		}
+
+		outputCarPath := cctx.Args().Get(2)
+
+		ctx := lcli.ReqContext(cctx)
+
+		// setup libp2p host
+		log.Infow("generating libp2p key")
+		privKey, _, err := crypto.GenerateECDSAKeyPair(rand.Reader)
+		if err != nil {
+			return err
+		}
+
+		host, err := libp2p.New(
+			libp2p.Transport(tcp.NewTCPTransport),
+			libp2p.Transport(quic.NewTransport),
```

Comment:
Are these the only transports you're supporting because they're the only ones Boost supports?